### PR TITLE
Splines linear problem pds tridiag

### PR DESCRIPTION
**Pull request by blegouix**

### CODE REVIEW

File: tests/splines/matrix.cpp [hunk 1]
```diff
@@ -201,6 +201,28 @@ TEST(Matrix, PDSBand)
...
 
+TEST(Matrix, PDSTridiag)
+{
+    std::size_t const N = 10;
+    std::size_t const k = 1;
+    std::unique_ptr<ddc::detail::SplinesLinearProblem<Kokkos::DefaultExecutionSpace>> matrix
+            = std::make_unique<
+                    ddc::detail::SplinesLinearProblemPDSTridiag<Kokkos::DefaultExecutionSpace>>(N);
+
+    // Build a positive-definite symmetric full-rank tridiagonal matrix
```
Comment (blegouix) [inline tests/splines/matrix.cpp:212]: positive-definiteness has been checked

**Review by tpadioleau — APPROVED** [review]
Looks ok to me. A minor concern about the layout of m_q. It has only one valid layout compared to the other classes.

**Comment by blegouix** [comment]
Yes that's true, LayoutRight is mandatory here.

I need reapproval.

**Review by tpadioleau — APPROVED** [review]

**Comment by yasahi-hpc** [comment]
I have a question about this PR. What is a typical storage format for this type of matrix. It seems that they have two separate matrices for diag and non-diag in [lapack](https://netlib.org/lapack/explore-html/d7/d77/dpttrs_8f_source.html).

**Comment by tpadioleau** [comment]
Indeed we have made a different choice. I think most libraries are inspired by the original design of LAPACK which uses 3 different pointeurs to store the diagonals in the general tridiagonal solver.

**Comment by yasahi-hpc** [comment]
Sure. I guess I need to employ the `lapack` interface

**Pull request merged**